src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Miner.Interfaces;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Features.Wallet.Controllers;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.IntegrationTests.Builders;
+using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+using Xunit.Abstractions;
+using static Stratis.Bitcoin.Features.Miner.PosMinting;
+
+namespace Stratis.Bitcoin.IntegrationTests.Miners
+{
+    public partial class ProofOfStakeMintCoinsSpecification
+    {
+        private IDictionary<string, CoreNode> nodes;
+        private NodeGroupBuilder nodeGroupBuilder;
+        private const string PowMiner = "ProofOfWorkNode";
+        private const string PosStaker = "ProofOfStakeNode";
+        private SharedSteps sharedSteps;
+
+        private HdAddress powSenderAddress;
+        private Key powSenderPrivateKey;
+
+        private HdAddress posReceiverAddress;
+        private Key posReceiverPrivateKey;
+
+        private const string PowWallet = "powwallet";
+        private const string PowWalletAccount = "account 0";
+        private const string PowWalletPassword = "password";
+
+        private const string PosWallet = "poswallet";
+        private const string PosWalletAccount = "account 0";
+        private const string PosWalletPassword = "password";
+
+        public ProofOfStakeMintCoinsSpecification(ITestOutputHelper outputHelper) : base(outputHelper)
+        {
+        }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+            this.nodeGroupBuilder = new NodeGroupBuilder();
+        }
+
+        protected override void AfterTest()
+        {
+            this.nodeGroupBuilder.Dispose();
+        }
+
+        private void a_proof_of_work_node_with_wallet()
+        {
+            this.nodes = this.nodeGroupBuilder
+                    .CreateStratisPowMiningNode(PowMiner)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PowWallet, PowWalletPassword)
+                    .Build();
+
+            this.powSenderAddress = this.nodes[PowMiner].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PowWallet, PowWalletAccount));
+            var wallet = this.nodes[PowMiner].FullNode.WalletManager().GetWalletByName(PowWallet);
+            this.powSenderPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PowWalletPassword, this.powSenderAddress).PrivateKey;
+        }
```

Comment:
The above three lines are repeated further down, maybe move into it's own method.

Reply:
@ferdeen they are similar but they set different variables ;)